Automattic/wp-calypso
# Temporary fix to optimize grid performance by reducing renders based on the deef dip of props.

* Builds on top of https://github.com/Automattic/wp-calypso/pull/83554
* More context: pdvytD-oN-p2
* Related to: Automattic/dotcom-forge#3984

## Proposed Changes

- While we race to reduce the data inconsistencies and memoize our data transformations. This change attempts to supplement those efforts by selectively rendering one of the most heaviest parts of the grid based on whether actual difference in props is visible.
- We also prevent rendering whenever the comparison grid is hidden from site to boost performance even more.
- The usage of context was moved to the root of the comparison grid, resulting in some prop drilling. However this is essential to prevent the leaking  of prop changes bypassing the memoized component.

## Testing Instructions



* Make sure the comparison grid renders correctly in various settings and flows (`/plans,`/start/plans`, `/setup/plans`)

## Pre-merge Checklist
